activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,16 +45,20 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
+      raise RuntimeError, "Anonymous class is not allowed." unless name
```

Comment:
I dunno, I moved that line from the connection_handler class to here, to keep the behavior, so I didnt wanted to change it.